### ISSUE TITLE
Feature request: support publishing Maven plugin documentation

### ISSUE THREAD

**Issue by ascopes** [maxcellent/javadoc.io#190]
Not sure if this is the right forum or whether this is already supported. I was hoping to raise this as a discussion rather than an issue but it appears that the tab is disabled for that here.

I have developed a Maven plugin. Right now, I have only got GitHub pages available to host the site generated by `./mvnw site`. This is mostly fine but means I cannot easily publish older versions of my documentation permanently.

I am wondering if it would be possible to potentially support me uploading my Maven site within a JAR to Maven Central similar to how JavaDocs are uploaded to there, or failing that allow some kind of API endpoint I can POST a generated site to so that I can make use of javadoc.io to publish the Maven plugin documentation. 

Would this be considered reasonable, do you think?

**Comment by maxcellent** [comment]
hi, the maven plugin you are using, is it for central maven repo? If that's the case it is automatically sync to javadoc.io already

**Comment by ascopes** [comment]
yep it is! Do I just visit the URL matching the GAV in that case, or do I need to configure the site artifacts?

Thanks for the quick response

**Comment by maxcellent** [comment]
yes, I believe visiting the URL is enough. Please let me know if this is not the case

**Comment by ascopes** [comment]
So I've just double checked... it appears to be pointing at the JavaDocs themselves rather than the output of `./mvnw site`, which was what I was mostly expecting.

Is there a way to publish that output in place of this?

**Comment by ascopes** [comment]
@maxcellent sorry to bother, but is there any update on how to work around this, or how to package Maven Site output in such a way it is handled correctly? It is difficult to test this without making actual releases of my software.

**Comment by maxcellent** [comment]
hi, do you have a sample library for me to check?

**Comment by ascopes** [comment]
https://github.com/ascopes/protobuf-maven-plugin/tree/main

```
./mvnw clean site -Dmaven.test.skip -Dinvoker.skip
```

Requires JDK 11 through 23.

Thanks for taking a look. It is greatly appreciated.
